### PR TITLE
[SPARK-40717][CONNECT] Support Column Alias in the Connect DSL

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Support Column Alias in the Connect DSL (thus in Connect proto).

### Why are the changes needed?

Column alias is a part of dataframe API , meanwhile we need column alias to support `withColumn` etc. API.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @cloud-fan

File: connector/connect/src/main/protobuf/spark/connect/expressions.proto [hunk 1]
```diff
@@ -165,4 +166,9 @@ message Expression {
   message QualifiedAttribute {
     string name = 1;
   }
+
+  message Alias {
```
Comment (grundprinzip) [inline connector/connect/src/main/protobuf/spark/connect/expressions.proto:170]: why wouldn't this be a common attribute of Expression instead?

Comment (cloud-fan) [reply connector/connect/src/main/protobuf/spark/connect/expressions.proto:170]: Only a few places require Expression to be wrapped by an alias. I'd like to match catalyst and have an individual `Alias` expression.

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

File: connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala [hunk 2]
```diff
@@ -81,6 +81,15 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
...
   }
 
+  test("column alias") {
+    val connectPlan = {
+      import org.apache.spark.sql.connect.dsl.expressions._
+      import org.apache.spark.sql.connect.dsl.plans._
+      transform(connectTestRelation.select("id".protoAttr.as("id2")))
+    }
+    val sparkPlan = sparkTestRelation.select($"id".as("id2"))
+  }
```
Comment (HyukjinKwon) [inline connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala:91]: Actually this doesn't test :-). Let me make a quick followup.

**Pull request closed without merging**